src/stan/mcmc/hmc/nuts/adapt_dense_e_nuts.hpp

```diff
@@ -33,6 +33,7 @@ class adapt_dense_e_nuts : public dense_e_nuts<Model, BaseRNG>,
           this->z_.inv_e_metric_, this->z_.q);
 
       if (update) {
+        this->metric_check(this->z_.inv_e_metric_.allFinite());
```

Comment:
The name here is misleading -- `check_metric` should be changed to something like `check_metric_condition` or just accept the inverse metric by reference and check allFinite itself.

Reply:
True enough. I wasn't sure what to call it--I just wanted to put error message in one place. Most logical would be to accept the metric by reference but `diag_e` and `dense_e` are `Eigen::VectorXd` and `Eigen::MatrixXd` and I'm not sure you can mix those...I guess it could be a virtual method on `ps_point`?

Reply:
That would be the most natural for the current OOP design.